pkg/operator/configobserver/node/observe_latency_profile.go

```diff
@@ -0,0 +1,138 @@
+package node
+
+import (
+	"reflect"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// LatencyProfileArgVal is used to set arg value pairs for each worker latency profile
+type LatencyProfileArgVal struct {
+	// path to required argument in observed config
+	// eg. observedConfig.extendedArguments.node-monitor-grace-period
+	// is []string{"extendedArguments", "node-monitor-grace-period"}
+	ArgConfigNestedPath []string
+	// value that is to be set for the given argument
+	ArgValue string
+}
+
+// CreateLatencyProfileObserverFn is used to create ObserveConfigFunc that can be used with an configobservation controller to trigger
+// changes to different arg val pairs in observedConfig.* fields and update them on the basis of current worker latency profile
+func CreateLatencyProfileObserverFn(latencyProfileValMap map[configv1.WorkerLatencyProfileType][]LatencyProfileArgVal, conditionFn func() (bool, error)) configobserver.ObserveConfigFunc {
+	observeLatencyProfileFn := func(
+		genericListers configobserver.Listers,
+		eventRecorder events.Recorder,
+		existingConfig map[string]interface{},
+	) (ret map[string]interface{}, errs []error) {
+
+		// traverse among different worker latency profiles and their required arg val pairs
+		// and obtain a unique list of paths that are to be used with the observedConfig
+		uniqueNestedPaths := getUniqueNestedPaths(latencyProfileValMap)
+
+		defer func() {
+			// Prune the observed config so that it only contains fields specific to this observer
+			ret = configobserver.Pruned(ret, uniqueNestedPaths...)
+
+			// in case of existingConfig and currently observedConfig has mismatch in fields that are
+			// being monitored eg. apiServerArguments.default-not-ready-toleration-seconds,
+			// extendedArguments.node-monitor-grace-period, etc.
+			prevRet := configobserver.Pruned(existingConfig, uniqueNestedPaths...)
+			if !reflect.DeepEqual(prevRet, ret) {
+				eventRecorder.Eventf("LatencyProfileUpdated", "latency profile on cluster was updated, observed-config = %v", ret)
+			}
+		}()
+
+		listers := genericListers.(NodeLister)
+		configNode, err := listers.NodeLister().Get("cluster")
+		// we got an error so without the node object we are not able to determine worker latency profile
+		if err != nil {
+			// if config/v1/node/cluster object is not found, that can be treated as a non-error case
+			if apierrors.IsNotFound(err) {
+				klog.Warningf("nodes.config.openshift.io/cluster not found")
```

Comment:
since it can be treated as a `non-error` case it will spam the logs, I'd remove it.

Reply:
Added it in favour of https://github.com/openshift/cluster-kube-controller-manager-operator/pull/611#discussion_r859543150.

Should it not be fine to see that in the logs upon every re-sync (though there would be too many repeated entries, I am not entirely sure how we could remove repetition)?

Reply:
Updated (removed warningf).